unfoldingWord/translation-helps-rcl
# Feature 399a - Merge from Master Branch-Merger RCL

# Describe what your pull request addresses

- [ ] Replace current merging from master with Branch-Merger RCL created by Abel

## Test Instructions

- [ ] Open [gEdit deploy preview](https://deploy-preview-434--gateway-edit.netlify.app/) and make sure you are on QA. 

- [ ] Change the organization in account settings to test_org or test_org2 for ease of resource editing. 
- [ ] Make edits in both a scripture card and a resource card so that a user branch is created. 
- [ ] In DCS, open the resource that you have edited and make a change. Upload this directly to master
- [ ] Go back to gEdit and reload to verify that these changes are ready to update from master. You should see a badge on both the icon in the resource and the icon in the header.

![image](https://github.com/unfoldingWord/gateway-edit/assets/118755839/2854aedd-d055-4880-91d2-5c172ce811ff)
Icons with badges to signal update. There are three different states of the icon to verify...
1. Greyed out with slash through clock - Conflict or error exists OR there is no user branch
2. Greyed out clock - User branch for this resource exists but there are no changes from master
3. Black clock with badge - User branch exists and there are someone else's updates to pull

- [ ] Click the icon (either header for all or individual for individual resources) and make sure the resources update accordingly.
- [ ] Verify other states of the application...
  - [ ] If no changes,  to pull when clicking header merge button, a dialog should appear alerting you of.
  - [ ] Cards with no updates
  - [ ] Cards with errors
  - [ ] Cards with conflicts

![image](https://github.com/unfoldingWord/gateway-edit/assets/118755839/d6a5d4ce-c281-48ec-aebf-4c138bfd259f)
Here is an example of the modal when clicking on the HEADER button with no changes...

- [ ] Individual Card merge buttons should also display a dialog when no changes are ready to pull...

![image](https://github.com/unfoldingWord/gateway-edit/assets/118755839/8632c247-d424-491f-980a-b04c1b7301b3)
